devicetypes/smartthings/cooper-scene-controller.src/cooper-scene-controller.groovy

```diff
@@ -0,0 +1,268 @@
+	//Not supported
+	//We have no way to set and/or retrieve multilevel state of each button
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStopLevelChange cmd) {
+	//Not supported
+	//We have no way to set and/or retrieve multilevel state of each switch
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationBusy cmd) {
+	//we have no way of knowing which command was ignored
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.scenecontrollerconfv1.SceneControllerConfReport cmd) {
+	if (cmd.groupId != cmd.sceneId) {
+		//Scene not set up properly for this association group. Try setting it two more times.
+		def cmds = []
+		cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfSet(dimmingDuration: 0, groupId: cmd.groupId, sceneId: cmd.groupId).format())
+		cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfSet(dimmingDuration: 0, groupId: cmd.groupId, sceneId: cmd.groupId).format())
+		sendHubCommand(cmds, 1500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unexpected zwave command $cmd"
+}
+
+//method created to make child class reusable
+void childOn(deviceNetworkId) {
+	def switchId = deviceNetworkId?.split("/")[1]
+	//this may override other local switch states if cloud is out of sync
+	updateLocalSwitchState()
+}
+
+//method created to make child class reusable
+void childOff(deviceNetworkId) {
+	def switchId = deviceNetworkId?.split("/")[1]
+	//this may override other local switch states if cloud is out of sync
+	updateLocalSwitchState()
+}
+
+private setSwitchState(switchId, state) {
+	String childDni = "${device.deviceNetworkId}/${switchId}"
+	def child = childDevices.find{it.deviceNetworkId == childDni}
+	if (!child) {
+		log.error "Child device $childDni not found"
+	}
+	//send event only if state changed
+	if(child?.device.currentState("switch")?.value != state) {
+		child?.sendEvent(name: "switch", value: "$state", descriptionText: "$child.displayName was switched $state")
+	}
+}
+
+private updateLocalSwitchState() {
+	def binarySwitchState = 0;
+	def multiplier = 1;
+	for (int i = 1; i <= 5; ++i) {
+		String childDni = "${device.deviceNetworkId}/${i}"
+		def child = childDevices.find{it.deviceNetworkId == childDni}
+		if (child?.device.currentState("switch")?.value == "on") {
+			binarySwitchState += multiplier
+		}
+		multiplier *= 2
+	}
+	sendHubCommand new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: binarySwitchState).format())
+}
+
+private addChildSwitches() {
+	for (i in 1..5) {
+		String childDni = "${device.deviceNetworkId}/${i}"
+		def child = childDevices.find{it.deviceNetworkId == childDni}
+		addChildDevice("Child Switch", "${childDni}", null,
+				[completedSetup: true, label: "${device.displayName} switch ${i}",
+				 isComponent: true, componentName: "switch$i", componentLabel: "Switch $i"])
```

Comment:
Not a big deal, but there's a lot of unnecessary closures here.

`"${childDni}"` is already a string and doesn't need to be substituted

`${i}` doesn't need a closure

Reply:
Thanks, I will fix those.